### ISSUE TITLE
strange design when wallet widget is minimized

### ISSUE THREAD

**Issue by herrhelms** [breadwallet/breadwallet#29]
Is this the intended behavior/design for the widget? See screenshot below:

![minimized](https://cloud.githubusercontent.com/assets/4956819/19784764/7a8e5190-9c86-11e6-90e6-6a88046a8925.jpg)

Running on iPhone7/10.1 breadwallet v0.6.4-46

**Comment by voisine** [comment]
That's strange, I also have an iPhone7/10.1 and mine looks like this:

![img_0114](https://cloud.githubusercontent.com/assets/316879/19785519/809543e2-9c4e-11e6-942a-f9b9c5cbc021.PNG)

Does tapping Show More/Show Less, or removing and re-adding the widget make any difference?

**Comment by herrhelms** [comment]
I tried but still the same thing

**Comment by voisine** [comment]
Ah, I bet it's because it's using the QR image cached from the previous version of the widget. When you receive funds and it will generate a new one for the new address that I think is a little larger.

**Comment by herrhelms** [comment]
Interesting train of thought. Okay, that would make sense in my case. So just a matter until someone wires some satoshis 😂 LOL

**Comment by voisine** [comment]
In the interest of science...

**Comment by herrhelms** [comment]
Yeah to science! Thanks for that! Here's what happened: QR code/address within app changed. I swiped to see the Widget, the old QR was shown for a second then replaced with the new one. - Same 'broken' layout though, and I did remove the widget, and pressed show less/more a couple of times. Should I reimburse you?

**Comment by voisine** [comment]
Darn... so strange that I can't reproduce it on the same device and OS version.

No you can keep the buck... don't spend it all in one place.

**Comment by herrhelms** [comment]
I just took a quick look into the repo - Some weird constrains behaving mysteriously upon messing with self.preferredContentSize - I will try to tinker with it a little over the weekend. The minified widget has some room for improvement, right?

**Comment by voisine** [comment]
yes, I quickly reworked it to get something that didn't look like ass with the new iOS 10 widgets, and maybe didn't succeed.

**Comment by voisine** [comment]
If you're feeling ambitious and want to re-do it in swift 3, then it would likely make it into the swift 3 rewrite we're doing.

**Comment by BeauNouvelle** [comment]
@voisine Where can we help contribute to the Swift version?
